File: Domain/Entities/StockProduct/ItemProduct.cs
using System;
using Domain.Entities.Shared;

using Domain.Entities.Main;
using Domain.Entities.Authentication;
using Domain.Entities.Main.Customers;

namespace Domain.Entities.StockProduct
{
    public class ItemProduct : RootBase
    {
        public int StockId { get; set; }
        public Stock Stock { get; set; }
        public int? IsReservedByUserId { get; set; } = null;
        public MyUser IsReservedByUser { get; set; }
        public DateTime IsReserved { get; set; }
        public int? ReservedForCustomerId { get; set; } = null;
        public Customer ReservedForCustomer { get; set; }
        public int SupplierId { get; set; }
        public Partner Supplier { get; set; }
        public string UsedHistoricalOrSupplier { get; set; }
        public string PurchaseInvoiceNumber { get; set; } // Número de Nfe de entrada
        public decimal CostPrice { get; set; }
        public decimal SoldPrice { get; set; }
        public DateTime EntryDate { get; set; }
        public DateTime SoldDate { get; set; }
        public DateTime WarrantyEnd { get; set; }
        public DateTime WarrantyEndLocal { get; set; }
        public bool IsUsed { get; set; } = false;
        public DateTime IsTested { get; set; } = DateTime.MinValue;
        

    }
}

File: Views/src/components/stock-product/product/dtos/item-product.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { SegmentDto } from "./segment-dto";
import { PartnerDto } from "src/components/main/partner/commons-components/dtos/partner-dto";
import { StockDto } from "./stock-dto";
import { MyUser } from "src/components/authentication/dto/my-user";
import { CustomerDto } from "src/components/main/customer/components/commons-components/dtos/customer-dto";

export class ItemProductDto extends RootBase {
    
    stockId: number;
    stock: StockDto;
    isReservedByUserId: number;
    isReservedByUser: MyUser;
    isReserved: Date;
    reservedForCustomerId: number;
    reservedForCustomer: CustomerDto;
    supplierId: number;
    supplier: PartnerDto;
    usedHistoricalOrSupplier: string;
    purchaseInvoiceNumber: string;
    costPrice: number;
    soldPrice: number;
    entryDate: Date;
    soldDate: Date;
    warrantyEnd: Date;
    warrantyEndLocal: Date;
    isUsed: boolean;
    isTested: Date;
}